gdpr/impl.go

```diff
@@ -135,7 +135,11 @@ func (p *permissionsImpl) allowActivities(ctx context.Context, vendorID uint16,
 
 	// vendor will be nil if not a valid TCF2 consent string
 	if vendor == nil {
-		return false, false, false, nil
+		if weakVendorEnforcement && parsedConsent.Version() == 2 {
```

Comment:
Could we fix this by changing `parseVendor` to return an error when the version is incompatible? Otherwise, this is fine for now and @bsardo could keep it in mind for refactoring.

Reply:
I went with this method to provide the least possible change in behavior. But we could start pushing back a bad TCF version when this happens. Making the change would also mean pushing `weakVendorEnforcement` back down a level again.

Reply:
I'm good with this change for now. I can make a note on the refactor story to return an error when the version is incompatible.